src/org/thoughtcrime/securesms/WelcomeActivity.java

```diff
@@ -355,6 +362,13 @@ public void handleEvent(@NonNull DcEvent event) {
 
         if (eventId== DcContext.DC_EVENT_IMEX_PROGRESS ) {
             long progress = event.getData1Int();
+            if (progressDialog == null || notificationController == null) {
+                // IMEX runs in BackupTransferActivity
+                if (progress == 1000) {
+                    finish();  // transfer done - remove ourself from the activity stack (finishAffinity is available in API 16, we're targeting API 14)
+                }
+                return;
+            }
```

Comment:
Is my understanding correct here?

So, when we do IMEX in BackupTransferActivity, the WelcomeActivity stays in the background so that the user can go back if something doesn't work, and when it's done, we call finish().

But then, why do we need `progressDialog == null || notificationController == null`? Wouldn't `progressDialog == null` be enugh?

Reply:
maybe, it would be sufficient, yes. it is just to be safe in case there are some maybe weird code paths followed before by user tapping around widely :) user may have eg. started + aborted backup or configure before - and tbh, i cannot oversee all cases :)